sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/implementation/SessionNotAvailableRetryTest.java

```diff
@@ -0,0 +1,338 @@
+        ServerStoreModel storeModel = (ServerStoreModel) FieldUtils.readField(rxDocumentClient, "storeModel", true);
+        StoreClient storeClient = (StoreClient) FieldUtils.readField(storeModel, "storeClient", true);
+        ReplicatedResourceClient replicatedResourceClient =
+            (ReplicatedResourceClient) FieldUtils.readField(storeClient, "replicatedResourceClient", true);
+        ConsistencyReader consistencyReader = (ConsistencyReader) FieldUtils.readField(replicatedResourceClient,
+            "consistencyReader", true);
+        ConsistencyWriter consistencyWriter = (ConsistencyWriter) FieldUtils.readField(replicatedResourceClient,
+            "consistencyWriter", true);
+        StoreReader storeReader = (StoreReader) FieldUtils.readField(consistencyReader, "storeReader", true);
+
+        RntbdTransportClientTest rntbdTransportClient = new RntbdTransportClientTest();
+        RntbdTransportClientTest spyRntbdTransportClient = Mockito.spy(rntbdTransportClient);
+        FieldUtils.writeField(storeReader, "transportClient", spyRntbdTransportClient, true);
+        FieldUtils.writeField(consistencyWriter, "transportClient", spyRntbdTransportClient, true);
+
+        cosmosAsyncContainer = getSharedMultiPartitionCosmosContainer(preferredListClient);
+
+        PartitionKey partitionKey = new PartitionKey("Test");
+        List<String> uris = new ArrayList<>();
+        doAnswer((Answer<Mono<StoreResponse>>) invocationOnMock -> {
+            Uri uri = invocationOnMock.getArgumentAt(0, Uri.class);
+            uris.add(uri.getURI().getHost());
+            CosmosException cosmosException = BridgeInternal.createCosmosException(404);
+            @SuppressWarnings("unchecked")
+            Map<String, String> responseHeaders = (Map<String, String>) FieldUtils.readField(cosmosException,
+                "responseHeaders", true);
+            responseHeaders.put(HttpConstants.HttpHeaders.SUB_STATUS, "1002");
+            FieldUtils.writeField(cosmosException, "responseHeaders", responseHeaders, true);
+            return Mono.error(cosmosException);
+        }).when(spyRntbdTransportClient).invokeStoreAsync(Mockito.any(Uri.class),
+            Mockito.any(RxDocumentServiceRequest.class));
+        try {
+            if (operationType.equals(OperationType.Read)) {
+                cosmosAsyncContainer.readItem("TestId", partitionKey, InternalObjectNode.class).block();
+            } else if (operationType.equals(OperationType.Query)) {
+                String query = "Select * from C";
+                CosmosQueryRequestOptions requestOptions = new CosmosQueryRequestOptions();
+                requestOptions.setPartitionKey(new PartitionKey("Test"));
+                cosmosAsyncContainer.queryItems(query, requestOptions, InternalObjectNode.class).byPage().blockFirst();
+            } else if (operationType.equals(OperationType.Create)) {
+                InternalObjectNode node = new InternalObjectNode();
```

Comment:
could you try to use POJO or ObjectNode for the new test? InternalObjectNode is internal type.

we are trying to remove the dependency on the InternalObjectNode specifically in the tests as might be used by customers as samples. so for new tests we should use POJO or ObjectNode.

Reply:
I would like to take that as separate work item , as i can see we are still using InternalObjectNode  in many test cases. We should just create one pojo and change all the test cases which is using InternalObjectNode .

Reply:
Naveen the ask is not to change the existing InternalObjectNode. The ask is we should not introduce more of these cases in the newly added test classes.

Reply:
Done